pkg/grpcfactory/server_test.go

```diff
@@ -141,5 +141,26 @@ var _ = Describe("gRPC Factory Server", func() {
 			// Clean up the socket file for future tests
 			os.Remove(socketPath)
 		})
+
+		It("should return an error for unsupported address schemes", func(ctx SpecContext) {
+			invalidAddress := "http://invalid-scheme-address" // Address with an unsupported scheme
+
+			server, err := grpcfactory.NewCOSIProvisionerServer(invalidAddress, identityServer, provisionerServer, nil)
+			Expect(err).NotTo(HaveOccurred()) // Ensure server creation succeeds
+			Expect(server).NotTo(BeNil())
+
+			errChan := make(chan error, 1)
```

Comment:
Wondering if we can just timeout this entire test case to simplify the way it's written. Perusing the gingko doc, I think `SpecTimeout` might do the trick but I'm not sure. If it applies, we could just synchronously wait for `server.Run` to return an error and check it. If it does not, the test will time out which is fine.

Reply:
Thats a great idea
Added an extra commit https://github.com/scality/cosi-driver/pull/56/commits/495c5ff1a128284cd4d52f664e96467ce91bb8ec